src/ensemble_scheduler.cc

```diff
@@ -1100,14 +1121,8 @@ EnsembleContext::CheckAndSetEnsembleOutput(
     }
   }
   if (!ready) {
-    if (info_->is_decoupled_) {
-      return Status::Success;
-    }
-    return Status(
-        Status::Code::INVALID_ARG,
-        lrequest->LogRequest() +
-            "unexpected deadlock, at least one output is not set while no more "
-            "ensemble steps can be made");
+    // Do not create response if the actual response is not ready.
```

Comment:
So this was partially related to original issue?  We were creating a response, when it was not ready?